code/__HELPERS/mobs.dm

```diff
@@ -295,6 +301,8 @@ GLOBAL_LIST_EMPTY(species_list)
 				break
 	if (progress)
 		qdel(progbar)
+	if(target)

```

Comment:
```suggestion
	if(!QDELETED(target))
```
You'll have to handle removing themselves from this list on `Destroy()`
You can use a signal for that.